### PR TITLE
Introduce barebone navigation components

### PR DESCRIPTION
**Pull request by catouberos**
## Describe your changes
Refactor were implemented for `AppButton` and `AppButtonLink`, includes:
- Changing `shadow-lg` to `shadow-md` on hover
- Remove unused child `div`
- `AppButtonLink` anchor was replace with Nuxt `<NuxtLink/>`

Components:
- `TheUser` component were introduced with barebone navigation to `/login`, `/register` and `/forgot` - not that pages were implemented yet
- `TheNavigation` component were introduced

### CODE REVIEW

File: components/TheUser.client.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+          <AppButtonLink
+            href="/register"
+            class="hover:bg-zinc-300 hover:dark:bg-zinc-600"
+          >
+            <Icon name="bi:person-plus"></Icon> Đăng ký
+          </AppButtonLink>
+
+          <AppButtonLink
+            href="/forgot"
```
Comment (Quang19992) [inline components/TheUser.client.vue:63]: why not  `/sign-out`

Comment (catouberos) [reply components/TheUser.client.vue:63]: it seems to be left out, mb

Comment (catouberos) [reply components/TheUser.client.vue]: resolved, @Quang19992 review again?

**Review by Quang19992 — APPROVED** [review]
LGTM

**Pull request merged**